File: 백준/Silver/1283. 단축키 지정/단축키 지정.swift
import Foundation

let N = Int(readLine()!)!
var shorts: [String] = []

for _ in 0..<N {
    let word = String(readLine()!)
    print(findShorts(word))
}

func findShorts(_ word: String) -> String {
    var wordList = word.components(separatedBy: " ")
    
    for (index, word) in wordList.enumerated() {
        let firstChar = String(word[word.startIndex]).uppercased()
        if !shorts.contains(firstChar) {
            let charList = word.map { String($0) }
            wordList[index] = makeShortOptionWord(charList, index: 0)
            shorts.append(firstChar)
            return wordList.joined(separator: " ")
        }
    }
    
    for (i, word) in wordList.enumerated() {
        let charList = word.map { String($0) }
        for (j, char) in charList.enumerated() {
            let check = char.uppercased()
            if !shorts.contains(check) {
                wordList[i] = makeShortOptionWord(charList, index: j)
                shorts.append(check)
                return wordList.joined(separator: " ")
            }
        }
    }
    return word
}

func makeShortOptionWord(_ word: [String], index: Int) -> String {
    var newWord = word
    newWord.insert("[", at: index)
    newWord.insert("]", at: index+2)
    return newWord.reduce(""){"\($0)\($1)"}
}

File: 백준/Silver/1283. 단축키 지정/단축키 지정.py
import sys
input = sys.stdin.readline

N = int(input())
shorts = []

def makeShortsOption(word, index):
    word.insert(index, "[")
    word.insert(index+2, "]")
    return "".join(word)
    
def findShorts(word):
    wordList = [w for w in map(str, word.split(" "))]
    
    for i in range(len(wordList)):
        firstChar = wordList[i][0].upper()
        if firstChar not in shorts:
            wordList[i] = makeShortsOption(list(wordList[i]), 0)
            shorts.append(firstChar)
            return " ".join(wordList)
            
    for i in range(len(wordList)):
        charList = list(wordList[i])
        for j in range(len(charList)):
            checkChar = charList[j].upper()
            if checkChar not in shorts:
                wordList[i] = makeShortsOption(charList, j)
                shorts.append(checkChar)
                return " ".join(wordList)
    
    return " ".join(wordList)

for _ in range(N):
    word = input().rstrip()
    result = findShorts(word)
    print(result)